### ISSUE TITLE
clickOnText action doesn't work

### ISSUE THREAD

**Issue by martinciu** [san650/ember-cli-page-object#106]
`clickOnText` creates selector with a white space before `:contains` for example: 

```javascript
# page object
clickAricle: clickOnText('h5');
```

```javascript
# test
page.clickArticle("Article 1");
```

generates selector:
`h5 :contains('Article 1'):last`
 but it should be: 
`h5:contains('Article 1'):last`

Is this test correct: https://github.com/san650/ember-cli-page-object/blob/master/tests/unit/actions/click-on-text-test.js#L10 ?

Selector like this doesn't work for me if I try it in the browser console.

**Comment by san650** [comment]
Hi @martinciu, Actually, the intention is to use the selector to scope the container in which to look for elements that contain that phrase. e.g. if you have a modal with buttons, you would do something like

```js
clickOn: clickOnText('.modal')
```

and then

```js
page.modal.clickOn('Submit');
```

The idea is that you scope the search by the element that matches the selector. If you already know the exact element you want to click on then you should use the `clickable` attribute.

Saying that, maybe there's an use case that I'm not seeing, please let me know.

Looking at the documentation [here](http://ember-cli-page-object.js.org/api/actions/#clickontext) I noticed that it actually says something similar to what you say :sweat_smile: , so we need to fix the documentation because it never worked that way.

**Comment by martinciu** [comment]
Hi @san650 , thanks for your reply.

This is my use case: I have several `h5` tags and I want to click only a few of them. For example:

```html
<h5>Article 1</h5>
<h5>Article 2</h5>
<h5>Article 3</h5>
<h5>Article 4</h5>
```

And I want to click Article 1 & 3. 

`clickable` won't work for this use case, will it?

**Comment by san650** [comment]
@martinciu no, it won't work for that use case.

You can create a [custom helper](http://ember-cli-page-object.js.org/api/components/#customhelper) though

```js
var clickTitle = customHelper(function(selector) {
  return function(text) {
    console.log(`Selector: ${selector}:contains("${text}")`);
    return click(`${selector}:contains("${text}")`);
  };
});

var page = PageObject.create({
  // ...
  title: clickTitle('h5')
});

// ... and in the test ....
page.visit().title('Log in');
```

Please, let me know if that works for you.

**Comment by mitchlloyd** [comment]
@san650 I just started using this library and I'm very impressed!

I also ran into this situation and was confused. I have some links like `a.class` and expected to use the `clickOnTextHelper` as well. Reading the documentation seems to enforce that this is a natural fit. 

I know this is a breaking change, but it seems like an improvement to remove the space. Then users can scope the selectors as they see fit: `.our-markup-needs-help p` or `.our-markup-needs-help p `.

**Comment by san650** [comment]
@mitchlloyd Thanks for the feedback. 

@mitchlloyd / @martinciu We could change the behavior of `clickOnText` to support current behavior and this new scenario in the 1.0 (see #91) release we're working right now. Let me think about the implementation a bit.

I'll fix the current documentation because 0.11 behaves differently as what is documented.

**Comment by san650** [comment]
@mitchlloyd Can you expand about what the `text` and `textList` inconsistencies are? Also, note that these two attributes are going to change in 1.0

**Comment by mitchlloyd** [comment]
@san650 I just looked at #91. Looks like a lot of exciting things are coming!

> Can you expand about what the text and textList inconsistencies are?

I was just noticing that `text` and `textList` take a selector that represents the node you're looking for (no space) whereas `clickOnText` takes a selector representing a node that surrounds the node that you're interested in.

**Comment by juanazam** [comment]
@mitchlloyd we included this change on `v1.0.0` do you think we can close this issue?

**Comment by mitchlloyd** [comment]
Just reading over the new docs, I like the solution! This appears consistent and is what I expect.

However, I saw one issue in the README where an old example was not updated for the new API. Fixed it here:

https://github.com/san650/ember-cli-page-object/pull/137

Once that is fixed, LGTM. Can't wait to try out the new version!

**Comment by juanazam** [comment]
We are very happy you like too, anyway there is more good stuff to come :smile:

**Comment by san650** [comment]
This is fixed in 1.0

Thanks everyone for all the help!
